Libraries/LLM/Configuration.swift

```diff
@@ -89,18 +90,44 @@ public struct ModelType: RawRepresentable, Codable, Sendable {
         },
     ]
 
-    public static func registerModelType(
-        _ type: String, creator: @escaping (URL) throws -> LLMModel
+    public func registerModelType(
+        _ type: String, creator: @Sendable @escaping (URL) throws -> LLMModel
     ) {
-        creators[type] = creator
+        lock.withLock {
+            creators[type] = creator
+        }
```

Comment:
by guarding the mutable state with the lock we can mark the type as Sendable (unchecked)